shell/platform/fuchsia/dart_runner/dart_runner.cc

```diff
@@ -92,6 +92,10 @@ void IsolateShutdownCallback(void* isolate_group_data, void* isolate_data) {
     tonic::DartMicrotaskQueue::GetForCurrentThread()->Destroy();
     async_loop_quit(loop);
   }
+
+  auto state =
+      static_cast<std::shared_ptr<tonic::DartState>*>(isolate_group_data);
```

Comment:
Not introduced in this PR, but passing around a `shared_ptr` via raw pointer seems fishy.  Is this because we are sending something to the Dart API's and then getting it sent back to us?

Reply:
Yes, this info is passed in when setting up the DartVM through the Dart API and is passed back to us here.

Reply:
Not for this PR but it would be safer to keep a `map<int, shared_ptr<tonic::DartState>>` and send the int to dart, then do a lookup when it gives it back to us instead of dereferencing a raw pointer.